src/Apps/Order/Routes/Shipping2/Components/AddressModal2.tsx

```diff
@@ -11,287 +10,396 @@ import {
+          />
+        }
+      </Formik>
+    </>
+  )
+}
 
-        if (updateAddressPayload?.__typename === "Errors") {
-          logger.error(
-            updateAddressPayload.errors.map(error => error.message).join(", ")
-          )
+const AddressModalForm: FC<{
+  onClose: () => void
+  addressModalAction: AddressModalProps["addressModalAction"]
+  onDeleteAddress: (address: SavedAddressType) => Promise<SavedAddressResult>
+}> = ({ addressModalAction, onClose, onDeleteAddress }) => {
+  const shippingContext = useShippingContext()
+  const formikContext = useFormikContext<FormValues>()
+  const [showDeleteDialog, setShowDeleteDialog] = useState<boolean>(false)
 
-          return
-        }
-      }
+  if (!addressModalAction) {
+    return null
+  }
 
-      onSuccess(savedAddressID)
-      setCreateUpdateError(null)
-      closeModal()
-    } catch (error) {
-      handleErrors([error], helpers)
-    } finally {
-      shippingContext.actions.setIsPerformingOperation(false)
+  const handleDeleteAddress = async () => {
+    if (addressModalAction?.type === "edit") {
+      try {
+        shippingContext.actions.setIsPerformingOperation(true)
+        await onDeleteAddress(addressModalAction.address)
+      } catch (error) {
+        logger.error(error)
+      } finally {
+        shippingContext.actions.setIsPerformingOperation(false)
+        setShowDeleteDialog(false)
+        onClose()
+      }
     }
   }
 
-  if (createUpdateError) {
-    logger.log({ createUpdateError })
+  const {
+    values,
+    touched,
+    errors,
+    handleChange,
+    handleBlur,
+    setFieldValue,
+  } = formikContext
+
+  const handleModalClose = () => {
+    formikContext.resetForm()
+    onClose()
   }
 
+  const title =
+    addressModalAction.type === "create" ? "Add address" : "Edit address"
+
   return (
     <>
       <ModalDialog title={title} onClose={handleModalClose} width={900}>
-        <Formik
-          validateOnMount
-          initialValues={initialAddress}
-          validationSchema={validationSchema}
-          onSubmit={handleSubmit}
-        >
-          {(formik: FormikProps<SavedAddressType>) => (
-            <form onSubmit={formik.handleSubmit}>
-              {createUpdateError && (
-                <Banner my={2} data-testid="form-banner-error" variant="error">
-                  {createUpdateError}
-                </Banner>
-              )}
-
-              <AddressModalFields />
-
-              <Spacer y={2} />
+        <Form data-testid="AddressModal">
+          {formikContext.status && (
+            <Banner my={2} data-testid="form-banner-error" variant="error">
+              {formikContext.status}
+            </Banner>
+          )}
 
+          <GridColumns mt={[1, 2]}>
+            <Column span={12}>
+              <Input
+                title="Full name"
+                placeholder="Full name"
+                id="name"
+                name="attributes.name"
```

Comment:
I copied these fields in so i could give them the `attributes` key nesting. On reflection, I'm not sure if this was necessary or was part of an earlier attempt to get mutation-calling consistent.